In [1]:
import pandas as pd
pd.set_option('display.max_columns', 50)
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sqlalchemy import create_engine
import scipy
from scipy import stats
import math as m

### Importing the table mk_effect.

This dataset contains the differences in average expenditures when a customer in part of a marketing campaign and when he is not. The purpose of this table is to do a Hypothesis contrast to see if there is any evidence to suggest that in fact customers are spending and visiting more when they are offered with a promotion, mailing, etc.

- H0: There is no differences in avg number of trips and avg expenditures when applying maketing campaigns. 
- HA: There is differences in avg number of trips and avg expenditures when the supermarket performs marketing campaigns.

In [2]:
mk_effect = pd.read_csv("../../data_final_project/mk_effect.csv")

In [3]:
mk_effect.head()

household_key  diff_avg_trips  diff_avg_spend  campaigns
0              1          0.2749        9.829216          8
1              2         -0.1663       -2.743128          1
2              3         -0.2499      -25.708983          3
3              4         -0.2353      -12.156438          1
4              6          0.6687        0.972535          4

In [4]:
mk_effect.shape

(1581, 4)

### Selecting the columns of interest to perfom the analysis.

- Difference in average trips or visits.
- Differences in average expenditures.

In [5]:
diff_avg_trips = list(mk_effect["diff_avg_trips"])
diff_avg_spend = list(mk_effect["diff_avg_spend"])

### Performing the confidence interval analysis on average trips.

In [6]:
n=len(diff_avg_trips)
ci = 0.95 
sample_std = np.std(diff_avg_trips)
quasi_std = sample_std*(m.sqrt(n/(n-1)))
sample_meam = np.mean(diff_avg_trips)

scipy.stats.t.interval(ci, n-1, loc=sample_meam, scale=quasi_std/m.sqrt(n))

print(f"We are 95% confident that the population mean lies between \
{scipy.stats.t.interval(ci, n-1, loc=sample_meam, scale=quasi_std/m.sqrt(n))}")

We are 95% confident that the population mean lies between (0.10223519158681858, 0.1914246439603035)


### Performing the confidence interval analysis on average expenditures. 

In [7]:
n=len(diff_avg_spend)
ci = 0.95 
sample_std = np.std(diff_avg_spend)
quasi_std = sample_std*(m.sqrt(n/(n-1)))
sample_meam = np.mean(diff_avg_spend)

scipy.stats.t.interval(ci, n-1, loc=sample_meam, scale=quasi_std/m.sqrt(n))

print(f"We are 95% confident that the population mean lies between \
{scipy.stats.t.interval(ci, n-1, loc=sample_meam, scale=quasi_std/m.sqrt(n))}")

We are 95% confident that the population mean lies between (5.673286683987849, 8.06836434812643)


#### Comments on the results:

In both cases, we can see that in 95% of the cases, the confidence intervals  do not include the 0. Based on this results, we can claim that there is statistical evidence to suggest that there is actually a differences when customers are offered a marketing campaign and when they are not. This step gives us evidence that in fact marketing efforts are effective in making the customers spend more and visiting more the supermarket.

## Getting the coefficient of variation to actual measure the effect in expenditures when customers are part of a campaign and when they are not. 

In [12]:
avg_in_mk = pd.read_csv("../../data_final_project/avg_after_mk.csv")

In [13]:
avg_off_mk = pd.read_csv("../../data_final_project/avg_before_mk.csv")

### Calculating the average expenditure when a customer is not part of a marketing campaign.

In [15]:
avg_off_mk.avg_spend.mean()

40.2072265900052

### Calculating the average expenditure when a customer is part of a marketing campaign.

In [14]:
avg_in_mk.avg_spend.mean()


47.10912781848704

### Calcualting the coefficient of variation to measure the effect of marketing campaigns.

In [10]:
((avg_in_mk.avg_spend.mean() / avg_off_mk.avg_spend.mean()) -1) *100

17.165822698642753

#### Comments on the results:

The results indicate that on average, customers spend 17% more when they are part of a marketing campaign than when they are not.

This give us a starting point to improve their marketing efforts by using data analytics techniques to leverage on data in order to perform customize campaigns thus, driving up revenues for the supermarket.  